components/mediation-ui/mediators-ui/org.wso2.carbon.mediator.enrich.ui/src/main/java/org/wso2/carbon/mediator/enrich/ui/EnrichMediator.java

```diff
@@ -272,13 +262,13 @@ public void build(OMElement elem) {
                 }
 
                 OMAttribute targetXpathAttr = childElem.getAttribute(new QName(XPATH));
-                if (targetXpathAttr != null
-                        && targetXpathAttr.getAttributeValue() != null
-                        && !targetXpathAttr.getAttributeValue().equals("")) {
+                if (targetXpathAttr != null && targetXpathAttr.getAttributeValue() != null && !targetXpathAttr
```

Comment:
Shall we get rid of the second null check `targetXpathAttr.getAttributeValue() != null` by having a `"".equals(targetXpathAttr.getAttributeValue())`

Reply:
We can't get rid of the second null check by this right? Still, we need to avoid the condition where targetXpathAttr.getAttributeValue() == null. This will only provide the fact that targetXpathAttr.getAttributeValue() is not empty.

Reply:
Yes, we can't, as Sasikala said second check for null value and `"".equals(targetXpathAttr.getAttributeValue())` for empty check.

Reply:
Agreed! Missed the ! before the equals

Reply:
yes `!("").equals(targetXpathAttr.getAttributeValue())`